### ISSUE TITLE
[elsa 2] send/dispatch signal not working anymore

### ISSUE THREAD

**Issue by vip32** [elsa-workflows/elsa-core#844]
Just updated to the latest preview7. Now the new DispatchSignalAsync (which replaces the old SendSignalAsync is assume) method does not really advance the workflow anymore.

The only change i noticed is that instead of the correlationid not the workflowinstanceid has to be supplied.
When i call the method like this:
```
await this.signaler.DispatchSignalAsync("approve", input: model, workflowInstanceId: instance.Id, cancellationToken: cancellationToken);
```

This is shown in the log. However the trigger does not really advance the workflow like it did before.
```
Mapped Elsa.Dispatch.TriggerWorkflowsRequest -> "TriggerWorkflowsRequest"
[15:17:59 INF] Starting periodic task "CleanupTrackedErrors" with interval 00:00:10
[15:17:59 INF] Bus "Rebus 2" setting number of workers to 1
[15:17:59 DBG] Adding worker "Rebus 2 worker 1"
[15:17:59 INF] Bus "Rebus 2" started
[15:17:59 DBG] Starting (threadpool-based) worker "Rebus 2 worker 1"
[15:17:59 DBG] Sending TriggerWorkflowsRequest { ActivityType = SignalReceived, Bookmark = Elsa.Activities.Signaling.SignalReceivedBookmark, Trigger = Elsa.Activities.Signaling.SignalReceivedBookmark, Input = Elsa.Activities.Signaling.Models.Signal, CorrelationId = , ContextId = , TenantId =  } -> "TriggerWorkflowsRequest"
```
what i see here ist that 'CorrelationId = , ContextId = , TenantId =' all have no values in the log, that could however be a logging issue.

Is the new DispatchSignalAsync still a work in progress or a bug?

**Comment by vip32** [comment]
also tried the following (TriggerSignalAsync)

```
await this.signaler.TriggerSignalAsync("approve", input: model, workflowInstanceId: instance.Id, cancellationToken: cancellationToken);
```

that either does'nt advance the workflow, and even less in the log (nothing from elsa)

**Comment by vip32** [comment]
@sfmskywalker is this a known issue for Elsa2? i am using the same triggering (TriggerSignalAsync) like in your normal trigger controller ([TriggerEndpoint](https://github.com/elsa-workflows/elsa-core/blob/74839211b83661e2f3e5ad7f7f7943c78073fabe/src/activities/Elsa.Activities.Http/Endpoints/Signals/TriggerEndpoint.cs#L30))

**Comment by sfmskywalker** [comment]
It could be a bug introduced with the recent refactorings, but I haven't been able to reproduce it. Perhaps if you could share with me your project or workflow (whichever is relevant to repro) than I will take a look.

**Comment by vip32** [comment]
yes must be a bug as it worked in earlier preview7, let me isolate something in a repo first...

**Comment by fubar-coder** [comment]
@craigfowler I have the same problem. You can test it easily with the `Elsa.Samples.SignalingConsole` sample project.

I see the following output when I start it:

```text
Car 1 is approaching red traffic light...
Car 2 is approaching red traffic light...
Hit enter to signal green light for Car 2.
```

I don't get any additional output when I press the ENTER/RETURN key.
